src/back/msl/writer.rs

```diff
@@ -25,6 +25,44 @@ const WRAPPED_ARRAY_FIELD: &str = "inner";
 // Some more general handling of pointers is needed to be implemented here.
 const ATOMIC_REFERENCE: &str = "&";
 
+/// Write the Metal name for a Naga numeric type: scalar, vector, or matrix.
+fn put_numeric_type(
+    out: &mut impl Write,
+    kind: crate::ScalarKind,
+    rows: Option<crate::VectorSize>,
```

Comment:
ooh, good idea

Reply:
Okay, I've implemented this. It's not quite perfect because you can't match on an `Option<VectorSize>` and produce a zero- or one-element slice: the array you're borrowing doesn't live long enough. But it's easy enough to just call `put_numeric_type` from each match arm instead.